### ISSUE TITLE
RPCA

### ISSUE THREAD

**Issue by Lrgrasso** [hyperspy/hyperspy#2277]
Hey guys, 
I was trying to perform robust principal componente analysis to remove outliers from some espectra data, but the things are not going the way I expected. 
For example, I have 250 spectrums, so I've loaded them with s = hs.signals.Signal1D(file) (navigation = 250; signal = 512). Then, used the function s.decomposition() and, with the scree plot, i choosed the output dimension. Then, I performed the RPCA (both algorithms) and the denoising function (s.get_decomposition_model()) with the output_dimension I've found before. However, when I retrieve the 'denoised' data, it is exactly the same thing it was before.
I could not figure out what I am doing wrong since I've followed all the steps in the documentation (http://hyperspy.org/hyperspy-doc/current/user_guide/mva.html), at least I think I have.
By the way, i tried the spikes_removal_tool(), it worked pretty well, but it's too slow for the amount of data i have.
Can you help me with that ?
 Thanks !!

**Comment by tjof2** [comment]
Can you post a code snippet with your parameters and use of `decomposition()`?

**Comment by tjof2** [comment]
Also note that both take some parameters, for example using GoDec you can try different values of `lambda1=X` where X is the regularization parameter.

Try values like 0.001, 0.01, 0.1, 1, 10 and see if that changes behaviour.

**Comment by Lrgrasso** [comment]
I performed, inicially, the PCA to determine the dimensions of the data:
`s = hs.signals.Signal1D(mat)`
`s.decomposition()`
`s.plot_explained_variance_ratio(vline = True)`
Then the RPCA:
`s.decomposition(algorithm="svd",output_dimension=6,return_info=True)`
`sc=s.get_decomposition_model(5)`
`(sc).plot()`

**Comment by tjof2** [comment]
You have algorithm=SVD?

As above, try a few different values for lambda1 as arguments to decomposition().

**Comment by Lrgrasso** [comment]
Oh, i was testing something and didn't realize i sent like this. I am using the RPCA_GoDec algorithm.

**Comment by tjof2** [comment]
Please try different values of lambda1 and post your code/results here

**Comment by tjof2** [comment]
To be clear, try something like
```
s = hs.signals.Signal1D(mat)
s.decomposition(algorithm="RPCA_GoDec", output_dimension=5, return_info=True, lambda=0.001)
sc=s.get_decomposition_model(5)
sc.plot()
```

**Comment by tjof2** [comment]
It occurred to me that there's a demo https://github.com/hyperspy/hyperspy-demos/blob/master/Online_Robust_PCA.ipynb here which shows it in full. Perhaps worth explicitly linking to this in the docs

**Comment by tjof2** [comment]
I think this can be closed - I'm improving the docs as part of #1159 and I think the problem here is solved by better docs.

**Comment by ericpre** [comment]
OK, let's close this PR once the doc have been improved!
